griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,293 @@
+        })
+    })
+    def save_memory_artifacts_to_drive(self, params: dict) -> ErrorArtifact | InfoArtifact:
+        values = params["values"]
+        memory = self.find_input_memory(values["memory_name"])
+        file_name = values["file_name"]
+    
+        if memory:
+            artifacts = memory.load_artifacts(values["artifact_namespace"])
+    
+            if not artifacts:
+                return ErrorArtifact("no artifacts found")
+
+            service = self._build_client(self.UPLOAD_FILE_SCOPES)
+
+            folder_id = None
+            if "folder_path" in values and values["folder_path"]:
+                folder_id = self._path_to_file_id(service, values["folder_path"])
+                if not folder_id:
+                    return ErrorArtifact(f"Could not find folder: {values['folder_path']}")
+
+            try:
+                if len(artifacts) == 1:
+                    self._save_to_drive(file_name, artifacts[0].value, folder_id)
+                else:
+                    for a in artifacts:
+                        self._save_to_drive(f"{a.name}-{file_name}", a.value, folder_id)
+    
+                return InfoArtifact(f"saved successfully")
+    
+            except Exception as e:
+                return ErrorArtifact(f"error saving file to Google Drive: {e}")
+        else:
+            return ErrorArtifact("memory not found")
+
+    @activity(config={
+        "description": "Can be used to save content to a file",
+        "schema": Schema(
+            {
+                Literal(
+                    "path",
+                    description="Destination file path on disk in the POSIX format. For example, 'foo/bar/baz.txt'"
+                ): str,
+                "content": str
+            }
+        )
+    })
+    def save_content_to_drive(self, params: dict) -> ErrorArtifact | InfoArtifact:
+        content = params["values"]["content"]
+        filename = params["values"]["path"]
+
+        try:
+            self._save_to_drive(filename, content)
+
+            return InfoArtifact(f"saved successfully")
+        except Exception as e:
+            return ErrorArtifact(f"error saving file to Google Drive: {e}")
+
+    def _save_to_drive(self, filename: str, value: any, parent_folder_id=None) -> InfoArtifact | ErrorArtifact:
+        service = self._build_client(self.UPLOAD_FILE_SCOPES)
+    
+        if isinstance(value, str):
+            value = value.encode()
+    
+        file_metadata = {"name": filename}
+        if parent_folder_id:
+            file_metadata["parents"] = [parent_folder_id]
+    
+        media = MediaIoBaseUpload(BytesIO(value), mimetype="application/octet-stream", resumable=True)
+    
+        try:
+            file = service.files().create(body=file_metadata, media_body=media, fields="id").execute()
+            return InfoArtifact(file.get("id"))
+        except HttpError as error:
+            return ErrorArtifact(f"An error occurred: {error}")
```

Comment:
Don't catch the exception here. Let the consumers do it.

Reply:
Done!